### PR TITLE
Fix #11416: Enter key behaviour on Extension Manager Search Box

### LINKED ISSUE #11416: 'Enter' key behavior on Extension Manager's Search Box
## How?

1. Open Extension Manager
2. Write something in the search box
3. Press 'Enter' to Search (but window closes)

## Why?

Because this is the user behavior when writing a search box.

Even without user-testing this, you can rest on well-known patterns that you use everyday: You write something on Chrome's address bar and press 'Enter' to go for it. You write on Google Search Box and you press 'Enter' to get results (also with live-search). Same here, on Github.

Even when the live-search feature is enable, the User could not be aware of this feature because of keypress timeouts, delayed server responses or even for looking at the keyboard while writing. So once he is done writing will press 'Enter' to send the request.

image

### PR DESCRIPTION
**Pull request by sahildua2305**
Fixes #11416
Made an exception for INPUT element in Dialogs.js so that we don't click
the primary button (in this case ESCAPE) when it's an INPUT.

### CODE REVIEW

File: src/widgets/Dialogs.js [hunk 1]
```diff
@@ -166,7 +166,9 @@ define(function (require, exports, module) {
             stopEvent();
             if (e.target.tagName === "BUTTON") {
                 this.find(e.target).click();
-            } else {
+            } else if (e.target.tagName != "INPUT") {
```
Comment (zaggino) [inline src/widgets/Dialogs.js]: by ESLint rules in place for project, `!==` is required instead of `!=`

Comment (sahildua2305) [inline src/widgets/Dialogs.js]: Fixed!

**Review by zaggino — APPROVED** [review]

**Comment by zaggino** [comment]
@petetnt @ficristo second look? just because you've been watching the issues reported around this probably more than I did

**Review by petetnt — APPROVED** [review]

**Comment by petetnt** [comment]
This looks good to me. Thanks for this contribution @sahildua2305!

**Pull request merged**